examples/audio_decoder.rs

```diff
@@ -0,0 +1,70 @@
+extern crate servo_media;
+
+use servo_media::audio::buffer_source_node::AudioBufferSourceNodeMessage;
+use servo_media::audio::decoder::AudioDecoderCallbacks;
+use servo_media::audio::node::{AudioNodeMessage, AudioNodeType};
+use servo_media::ServoMedia;
+use std::env;
+use std::fs::File;
+use std::io::Read;
+use std::sync::mpsc;
+use std::sync::{Arc, Mutex};
+use std::{thread, time};
+
+fn run_example(servo_media: Arc<ServoMedia>) {
+    let mut graph = servo_media.create_audio_graph();
+    let args: Vec<_> = env::args().collect();
+    let filename: &str = if args.len() == 2 {
+        args[1].as_ref()
+    } else {
+        panic!("Usage: cargo run --bin audio_decoder <file_path>")
+    };
+    let mut file = File::open(filename).unwrap();
+    let mut bytes = vec![];
+    file.read_to_end(&mut bytes).unwrap();
+    let decoded_audio: Arc<Mutex<Vec<f32>>> = Arc::new(Mutex::new(Vec::new()));
+    let progress = decoded_audio.clone();
+    let (sender, receiver) = mpsc::channel();
+    let callbacks = AudioDecoderCallbacks::new()
+        .eos(move || {
+            sender.send(()).unwrap();
+        })
+        .error(|| {
+            eprintln!("Error decoding audio");
+        })
+        .progress(move |buffer| {
```

Comment:
Is there another user for this already apart from the example? Some code that does not actually copy all the samples directly but reuses the slice?

Reply:
Not yet. I'm working on the DOM bindings using this code.